scripts/preinstall.sh

```diff
@@ -1,31 +1,59 @@
 #!/usr/bin/env bash
-# Check TensorFlow version
-c++ ./utils/checker/tensorflow_version_checker.cc -ltensorflow -o ./tf_ver
-./tf_ver
-if [ "$?" -eq "0" ]
+
+INSTALL_DIR=/usr/local
+INSTALL_SCRIPT=/tmp/install_script.sh
+TMP_DIR=/tmp/wasmedge
+mkdir -p "$TMP_DIR"
+ldconfig
+
+# Install WasmEdge
+if [ -z "$(ldconfig -p | grep libwasmedge_c.so)" ]
+then
+  wget -O "$INSTALL_SCRIPT" https://github.com/second-state/WasmEdge-go/releases/download/v0.8.1/install_wasmedge.sh
+  sh "$INSTALL_SCRIPT" "$INSTALL_DIR"
+  rm -f "$INSTALL_SCRIPT"
+fi
+
+# Install WasmEdge-tensorflow
+if [ -z "$(ldconfig -p | grep libtensorflow.so)" ]
+then
+  wget -O "$INSTALL_SCRIPT" https://github.com/second-state/WasmEdge-go/releases/download/v0.8.1/install_wasmedge_tensorflow_deps.sh
+  sh "$INSTALL_SCRIPT" "$INSTALL_DIR"
+  rm -f "$INSTALL_SCRIPT"
+fi
+
+if [ -z "$(ldconfig -p | grep libwasmedge-tensorflow_c.so)" ]
+then
+  wget -O "$INSTALL_SCRIPT" https://github.com/second-state/WasmEdge-go/releases/download/v0.8.1/install_wasmedge_tensorflow.sh
+  sh "$INSTALL_SCRIPT" "$INSTALL_DIR"
+  rm -f "$INSTALL_SCRIPT"
+fi
+
+# Install WasmEdge-image
+if [ -z "$(ldconfig -p | grep libpng)" -o -z "$(ldconfig -p | grep libjpeg)" ]
 then
-  rm ./tf_ver
-  exit 0
-else
-  echo "Error: TensorFlow C library is not yet installed or is too old to support"
-  exit 1
+  wget -O "$INSTALL_SCRIPT" https://github.com/second-state/WasmEdge-go/releases/download/v0.8.2/install_wasmedge_image_deps.sh
+  sh "$INSTALL_SCRIPT" "$INSTALL_DIR"
+  rm -f "$INSTALL_SCRIPT"
 fi
 
-# Check libjpeg-dev and libpng-dev
-ldconfig -p | grep libjpeg
-if [ "$?" -eq "0" ]
+if [ -z "$(ldconfig -p | grep libwasmedge-image_c.so)" ]
 then
-  exit 0
-else
-  echo "Error: libjpeg-dev is not installed"
-  exit 1
+  wget -O "$INSTALL_SCRIPT" https://github.com/second-state/WasmEdge-go/releases/download/v0.8.2/install_wasmedge_image.sh
+  sh "$INSTALL_SCRIPT" "$INSTALL_DIR"
+  rm -f "$INSTALL_SCRIPT"
 fi
 
-ldconfig -p | grep libpng
-if [ "$?" -eq "0" ]
+# Install WasmEdge-storage
+if [ -z "$(ldconfig -p | grep libwasmedge-storage_c.so)" ]
 then
-  exit 0
-else
-  echo "Error: libpng-dev is not installed"
-  exit 1
+  wget https://github.com/second-state/WasmEdge-storage/releases/download/0.8.1/WasmEdge-storage-0.8.1-Linux_x86_64.tar.gz
+  tar -C "$TMP_DIR" -xzf WasmEdge-storage-0.8.1-Linux_x86_64.tar.gz
+  mv "$TMP_DIR"/include/* "$INSTALL_DIR"/include
+  mv "$TMP_DIR"/lib/* "$INSTALL_DIR"/lib
+  rm -f WasmEdge-storage-0.8.1-Linux_x86_64.tar.gz
+  rm -rf "$TMP_DIR"/*
 fi
+
+ldconfig
```

Comment:
`ldconfig` will need root permission. Maybe we should have `sudo` here?

Reply:
added